### DISCUSSION TITLE
Python Version Numbering

### DISCUSSION THREAD

**Discussion by ykh-1992** [microsoft/winget-pkgs#40509] [General]
Hello.

I'm curious to know if there is a reason for the way Python is being numbered as 3.X.X150.0 and not just 3.X.X?
It was confusing as a newbie when I first tried to install Python through winget.

A somewhat similar case is when I requested a PR recently to update Rust to 1.57.0 but didn't quite understand why I had to rename it to 1.57.0.0 instead (even though the right most numbering is always zero).

Appreciate any feedback.

**Comment by vedantmgoyal9** [comment]
The string must match the one shown from Control Panel's Add or Remove Programs (ARP) list because the Windows Package Manager uses the strings and values shown from Control Panel's Add or Remove Programs (ARP) list to determine if there's an upgrade available using the `winget upgrade` command and to properly map the application in the `winget list` command.

**Comment by denelon** [comment]
This thread has a suggestion for what we're thinking with Python versions:
https://github.com/microsoft/winget-cli/discussions/794#discussioncomment-3182240
